File: javascript/pangram/pangram.js
const NUM_UNIQUE_CHARS = 26;

class Pangram {
  constructor(input) {
    this.originalSentence = input;
  }

  isPangram() {
    const cleanString = this.originalSentence.toLowerCase()
                                             .replace(/[^a-z]/g, '');

    return new Set(cleanString).size === NUM_UNIQUE_CHARS;
  }
}

module.exports = Pangram;

File: ruby/pangram/pangram.rb
module Pangram
  NUM_ALPHABET_CHARS = 26

  def self.is_pangram?(str)
    lower_case_chars_only = str.downcase.gsub(/[^a-z]/, '')
    num_unique_chars = lower_case_chars_only.chars.uniq.length
    num_unique_chars == NUM_ALPHABET_CHARS
  end
end

module BookKeeping
  VERSION = 2
end
